### DISCUSSION TITLE
RTK Query with micro frontends

### DISCUSSION THREAD

**Discussion by einarq** [reduxjs/redux-toolkit#2298] [General]
I'm currently using injectEndpoints in my app to lazy load parts of the app (code splitting), based on routes.
Now we are looking to take this one step further and share entire modules, including reducers and queries etc, across apps using module federation. The goal would be that each "module" that we federate would be able to inject its own reducers and queries into the store of the app they are loaded into.

Similar to the common approach suggested in the docs of exposing an injectReducer on the store you are injecting the module into, I was hoping to use the injectEndpoints approach to inject all of the endpoints of the module into the global RTK Query api object of the app you are loading the module into. Typing becomes a bit problematic though, as well as getting access to the auto-generated hooks. 

Has anyone seen any examples of how to do this?

**Comment by manoj-mukherjee-maersk** [comment]
Nextjs-mf we need redux-toolkit and RTK Query for MFE. Are any support for this?

**Reply by markerikson** [reply]
There's no specific support here, but no one has yet listed a specific problem you're trying to solve with "micro-frontends" either :)

If you can point to a very specific problem we can maybe suggest solutions, but just asking "is this supported?" isn't a question we can answer.

**Reply by manoj-mukherjee-maersk** [reply]
We trying to redux-toolkit with nextjs-mf csr and ssr. We need globalstore where multiple app register their store and from other app also dispatch event and listen the change. (RTK Query, Reducer). How https://github.com/microsoft/redux-micro-frontend/blob/main/src/global.store.ts trying to solve the problem but how we it SSR part?

**Reply by markerikson** [reply]
Sorry, I have no particular answers or suggestions here.  I've never tried to build a micro-frontend, and I haven't tried to do Redux SSR.  So, I'm not sure what the actual problems are that you're seeing, or what solutions might be.

**Comment by einarq** [comment]
Problem to solve:
A “shell” app of some sort, which pulls in modules from other apps, along with any reducers and query endpoints etc, and plugs those into the store of that shell app. And to have this built into RTK itself. 
Does that make it any clearer?

Perhaps similar to the old “Redux Modules” package from Microsoft?

**Reply by manoj-mukherjee-maersk** [reply]
@einarq any example how you solve - RTK query and Redux-toolkit MFE. We did normal redux dispatch and RTK query like usersAPI.endpoints.getUsersList.initiate way. When we trying use hooks its not working.

**Reply by markerikson** [reply]
@einarq out of curiosity, does https://github.com/fostyfost/redux-eggs help at all?
